src/core/frameworks/yearnSdk/index.ts

```diff
@@ -7,14 +7,20 @@ export class YearnSdkImpl implements YearnSdk {
   private instances: Map<Network, Yearn<SdkNetwork>> = new Map<Network, Yearn<SdkNetwork>>();
 
   constructor({ web3Provider, config }: { web3Provider: Web3Provider; config: Config }) {
-    const { SUPPORTED_NETWORKS } = config;
+    const { SUPPORTED_NETWORKS, YEARN_SUBGRAPH_ID, YEARN_SUBGRAPH_KEY } = config;
 
     SUPPORTED_NETWORKS.forEach((network) => {
       const providerType = getProviderType(network);
       const provider = web3Provider.getInstanceOf(providerType);
       const networkId = getNetworkId(network) as SdkNetwork;
       const sdkInstance = new Yearn(networkId, {
         provider,
+        ...(YEARN_SUBGRAPH_KEY && {
```

Comment:
we should have a ` || {}` at the end of this to send correct parameters?

Reply:
Is one of those js hacky things. Here some response I found https://stackoverflow.com/questions/64603492/i-dont-understand-about-spread-syntax-inside-objects